packages/dds/sequence/src/defaultMap.ts

```diff
@@ -327,14 +357,18 @@ export class DefaultMap<T> {
 	 * @returns True if the operation was processed, false otherwise.
 	 */
 	public tryProcessMessage(
-		op: IMapOperation,
+		op: unknown,
 		local: boolean,
-		message: ISequencedDocumentMessage | undefined,
+		message: ISequencedDocumentMessage,
 		localOpMetadata: unknown,
 	): boolean {
-		const handler = this.messageHandlers.get(op.type);
-		if (handler !== undefined) {
-			handler.process(op, local, message, localOpMetadata as IMapMessageLocalMetadata);
+		if (isMapOperation(op)) {
```

Comment:
no more dictionary. just check if its a map op via existing type guard, and if so, call the handler. eventually we can compress the abstraction here even further, but don't want this pr to introduce too much chrun